src/builders.ts

```diff
@@ -323,14 +346,30 @@ export async function makeSTXTokenTransfer(
       options.network.version === TransactionVersion.Mainnet
         ? AddressVersion.MainnetSingleSig
         : AddressVersion.TestnetSingleSig;
-    const senderAddress = publicKeyToAddress(addressVersion, pubKey);
+    const senderAddress = c32address(addressVersion, transaction.auth.spendingCondition!.signer);
     const txNonce = await getNonce(senderAddress, options.network);
     transaction.setNonce(txNonce);
   }
 
-  if (options.senderKey) {
+  if (options.multiSig) {
+    const signer = new TransactionSigner(transaction);
+    let pubKeys = options.multiSig.publicKeys;
+
+    for (const key of options.multiSig.signerKeys) {
```

Comment:
This would require all keys to be provided at the same time which I think is rarely the case when it comes to multi-sig wallets. The use case is likely to be similar to sponsored transactions, where a signed transaction is serialized and passed to the next signer and so on. 

I think it makes more sense to provide functions that allow transactions to be signed, serialized/deserialized and signed again.

Reply:
Makes sense!